src/plasma/malloc.cc

```diff
@@ -33,15 +35,10 @@ int fake_munmap(void *, size_t);
 
 struct mmap_record {
   int fd;
-  void *pointer;
   int64_t size;
-  UT_hash_handle hh_fd;
-  UT_hash_handle hh_pointer;
 };
 
-/* TODO(rshin): Don't have two hash tables. */
-struct mmap_record *records_by_fd = NULL;
-struct mmap_record *records_by_pointer = NULL;
+std::unordered_map<void *, mmap_record> mmap_records;
```

Comment:
You can put this in an unnamed namespace (http://en.cppreference.com/w/cpp/language/namespace#Unnamed_namespaces) to ensure that it's not accessible outside this .cc file (has the same effect as a static global variable in C/C++).